docs/DeveloperGuide.md

```diff
@@ -1,38 +1,253 @@
 # Developer Guide
 
-## Acknowledgements
+## Table of Contents
+
+1. [Acknowledgements](#acknowledgements)
+2. [Design](#design)
+   - [Architecture](#architecture)
+   - [UI component](#ui-component)
+   - [Parser component](#parser-component)
+   - [Storage component](#storage-component)
+   - [Common classes](#common-classes)
+3. [Implementation](#implementation)
+4. [Appendix: Requirements](#appendix--requirements)
+5. [Appendix: Instructions for manual testing](#appendix--instructions-for-manual-testing)
+
+---
 
+## Acknowledgements
+x
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
-## Design & implementation
+{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the
+original source as well}
```

Comment:
there seems to be duplicate sentences here, can consider updating and filling it up whenever possible!